### DISCUSSION TITLE
Issues with fixed position

### DISCUSSION THREAD

**Discussion by yzhang-github-pub** [facebookresearch/esm#380] [Show and tell]
To express inverse folding designed proteins, the first residue must be 'M' (for Methionine) which is encoded by start codon. So I want to fix the first residue. I tested the modified version of 'sample_sequences.py' by @martinpacesa from #236 on a short protein sequence of 86 residues (starting with 'M' which is the only 'M' in the sequence). 5 sequences were sampled using the original script, and the count of 'M' is [0, 1, 0, 1, 1]. Using the modified version, the count is [15, 1, 37, 31, 1]. Temperature is set to 0.5.

**Comment by chanjed** [comment]
I found similar issues with this (many repeated tokens) in different settings.

**Comment by tomsercu** [comment]
Interesting case study, thanks for sharing! Let's move this to discussions since this may be just a weird artefact of forcing that first token, or an issue with the forced sampling script. In either case not really anything actionable I can see from our side.

**Reply by yzhang-github-pub** [reply]
Thanks @tomsercu. I'd ask you to consider this as a feature request. The reason is that, for all practical purposes, the designed protein by esm-IF will be produced in cells, e.g., in E. coli. The protein sequence will be the template for DNA codons, and the first codon always encodes 'M' (Methionine). So it is necessary to fix the first token. In many cases it also makes sense to fix tokens in other positions. Please consider to add the feature allowing position fixing at arbitrary positions (and combinations of different positions).
